From 7797f83e1d66ff78872763e1da3a5fb2f0534c40 Mon Sep 17 00:00:00 2001
From: Markus Hennerbichler <markushennerbichler@gmail.com>
Date: Mon, 15 Jul 2024 14:07:13 +0100
Subject: [PATCH] Fix CUDA 12.5 build

CUDA 12.5 removed the FLT_MAX symbol.
This was previously used without being explicitly imported.
FLT_MAX is defined in <float.h>, including this header fixes the issue
---
 src/libtorchaudio/cuctc/src/ctc_prefix_decoder_kernel_v2.cu | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/libtorchaudio/cuctc/src/ctc_prefix_decoder_kernel_v2.cu b/src/libtorchaudio/cuctc/src/ctc_prefix_decoder_kernel_v2.cu
index 4ca8f1bf24..e6192155a2 100644
--- a/src/libtorchaudio/cuctc/src/ctc_prefix_decoder_kernel_v2.cu
+++ b/src/libtorchaudio/cuctc/src/ctc_prefix_decoder_kernel_v2.cu
@@ -24,6 +24,7 @@
 // (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
 // OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #include <algorithm>
+#include <float.h>
 #include "ctc_fast_divmod.cuh"
 #include "cub/cub.cuh"
 #include "device_data_wrap.h"
